titiler/stacapi/reader.py

```diff
@@ -82,6 +84,15 @@ def _get_asset_info(self, asset: str) -> AssetInfo:
         url = asset_info.get_absolute_href() or asset_info.href
         if alternate := stac_config.alternate_url:
             url = asset_info.to_dict()["alternate"][alternate]["href"]
+        
+        # No caching of this should be necessary. From the docs https://planetarycomputer.microsoft.com/docs/concepts/sas/#planetary-computer-python-package:
+        # A cache is also kept, which tracks expiration values, to ensure new SAS tokens are only requested when needed.
+        # We only want to sign requests to MS PC API. Other ways to handle this could be:
+        # 1. Check the asset's URL to see if contains 'blob.core.windows.net' (seems brittle)
+        # 2. Set a boolean in settings, something like "SIGN_REQUESTS"
+        # 3. Just assume all requests are to MS PC API and sign them. Stub out this function in tests and/or when ENV=test
+        if stac_api_config.stac_api_url == stac_api_config.mspc_default_api_url:
+            url = planetary_computer.sign(url)
```

Comment:
FYI this is how it's done in MC planetary computer https://github.com/microsoft/planetary-computer-apis/blob/main/pctiler/pctiler/reader.py#L43-L59

Reply:
to avoid having a new settings, we could simply check if the asset url is on MS datastore

Reply:
>FYI this is how it's done in MC planetary computer https://github.com/microsoft/planetary-computer-apis/blob/main/pctiler/pctiler/reader.py#L43-L59

Nice, thanks for sharing that. It looks like we would not use this code exactly because it relies on a `CollectionConfigTable`

>to avoid having a new settings, we could simply check if the asset url is on MS datastore

👍🏽 it does appear we can re-use some of https://github.com/microsoft/planetary-computer-apis/blob/main/pccommon/pccommon/cdn.py#L8 to do this. I noticed that this code determines if there is an azureedge configured for the data, but I don't think there is anyway for this application to know about that? Since that code relies on (I think) some redis cache? https://github.com/microsoft/planetary-computer-apis/blob/main/pccommon/pccommon/config/core.py#L28